### PR TITLE
Sync cloud settings

### PR DESCRIPTION
**Pull request by Algunenano**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Sync cloud settings

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-clickhouse** [comment]
*This is an automated comment for commit 0591dfba07224a7b521bb6123ab1526d1ff26556 with description of existing statuses. It's updated for the latest CI running*

[❌ Click here](https://s3.amazonaws.com/clickhouse-test-reports/72481/0591dfba07224a7b521bb6123ab1526d1ff26556/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Style check</td><td>Runs a set of checks to keep the code style clean. If some of tests failed, see the related log from the report</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/72481/0591dfba07224a7b521bb6123ab1526d1ff26556/style_check.html">❌ failure</a></td></tr>
<tbody>
</table>

File: tests/queries/0_stateless/02995_new_settings_history.sh [hunk 1]
```diff
@@ -17,7 +17,8 @@ $CLICKHOUSE_LOCAL --query "
     new_settings AS
     (
         -- Ignore settings that depend on the machine config (max_threads and similar)
-        SELECT name, default FROM system.settings WHERE default NOT LIKE '%auto(%'
+        -- Ignore settings that, for historic reasons, have different values in Cloud
+        SELECT name, default FROM system.settings WHERE default NOT LIKE '%auto(%' AND name NOT IN ('max_table_size_to_drop', 'max_partition_size_to_drop')
```
Comment (antaljanosbenjamin) [inline tests/queries/0_stateless/02995_new_settings_history.sh:21]: Where do `max_table_size_to_drop` and `max_partition_size_to_drop`? Where do they cause issues?

Comment (Algunenano) [reply tests/queries/0_stateless/02995_new_settings_history.sh:21]: They cause issues when enabling this test in private

Comment (Algunenano) [reply tests/queries/0_stateless/02995_new_settings_history.sh:25]: I'm going to rework this a bit to know exactly about CLICKHOUSE_CLOUD

**Review by antaljanosbenjamin — APPROVED** [review]

**Comment by antaljanosbenjamin** [comment]
The ASAN func tests failed with no credentials. I don't think your changes are connected to that failure, but it blocked a lot of tests, so I think we should figure that out first.

**Comment by Algunenano** [comment]
I'll update the PR to force the rerun

**Pull request merged**